libtransmission/rpc-server.c

```diff
@@ -685,18 +685,21 @@ static void handle_request(struct evhttp_request* req, void* arg)
 
         server->loginattempts = 0;
 
-        if (strncmp(req->uri, server->url, strlen(server->url)) != 0)
+        char* location = strstr(req->uri, server->url) ? req->uri + strlen(server->url) : NULL;
```

Comment:
Logically, `strstr` means "contains" and not "begins with". Although it's hard to imagine for both not to be the same in this particular case, it's still better to express your intent clearly; also, "contains" is less efficient since it needs to perform more comparisons in the worst case.

```suggestion
        size_t const server_url_len = strlen(server->url);
        char const* const location = strncmp(req->uri, server->url, server_url_len) == 0 ? req->uri + server_url_len : NULL;
```

As a side note, conditional expressions should be explicitly boolean: `strstr` return an `int`, not `bool`, which means a comparison operation (yielding boolean result) is missing.